src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java

```diff
@@ -127,37 +128,45 @@ public Navigation openRecentWikiActivity() {
     return this;
   }
 
-  public boolean isUserProfileLinkVisible() {
+  public boolean isMainHeaderVisible() {
     try {
-      return userProfile.isDisplayed();
+      return navigationMainHeader.isDisplayed();
     } catch (NoSuchElementException e) {
       PageObjectLogging.logInfo(e.getMessage());
       return false;
     }
   }
 
+  public boolean isBackButtonVisible() {
+    try {
+      return backButton.isDisplayed();
+    } catch (NoSuchElementException e) {
+      PageObjectLogging.logInfo(e.getMessage());
+      return false;
+    }
+  }
 
-  public boolean isMainHeaderVisible() {
+  public boolean isUserAvatarVisible() {
     try {
-      return navigationMainHeader.isDisplayed();
+      return userAvatar.isDisplayed();
     } catch (NoSuchElementException e) {
       PageObjectLogging.logInfo(e.getMessage());
       return false;
     }
   }
 
-  public boolean isBackButtonVisible() {
+  public boolean isUserProfileLinkVisible() {
     try {
-      return backButton.isDisplayed();
+      return userProfileLink.isDisplayed();
     } catch (NoSuchElementException e) {
       PageObjectLogging.logInfo(e.getMessage());
       return false;
     }
   }
 
-  public boolean isFooterVisible() {
+  public boolean isLogoutLinkVisible() {
     try {
-      return homeOfFandomFooter.isDisplayed();
+      return logoutLink.isDisplayed();
     } catch (NoSuchElementException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)